lib/mapping/MapFormatH3M.cpp

```diff
@@ -954,10 +954,11 @@ bool CMapLoaderH3M::loadArtifactToSlot(CGHeroInstance * hero, int slot)
 	// H3 bug workaround - Enemy hero on 3rd scenario of Good1.h3c campaign ("Long Live The Queen")
 	// He has Shackles of War (normally - MISC slot artifact) in LEFT_HAND slot set in editor
 	// Artifact seems to be missing in game, so skip artifacts that don't fit target slot
-	auto * artifact = ArtifactUtils::createArtifact(map, artifactID);
+	auto * artifact = ArtifactUtils::createArtifact(artifactID);
```

Comment:
I think this will now result in memory leak?
Since such artifact will not be added to map.
Probably needs delete in `else` block

Switching storage of artifacts in CMap to shared_ptr / unique_ptr might be better option, but is also way more complex

Reply:
No, I mean in this particular piece of code. Check several lines below.

Here `addNewArtifactInstance` is only called if condition on next line is true, but artifact itself is created unconditionally

Reply:
> Switching storage of artifacts in CMap to shared_ptr / unique_ptr might be better option, but is also way more complex

In todo list.